src/Events/Custom_Tables/V1/Schema_Builder/Schema_Builder.php

```diff
@@ -48,6 +48,37 @@ public function get_field_schemas_that_need_updates() {
 		return $this->filter_for_version( $this->get_registered_field_schemas() );
 	}
 
+	/**
+	 * Get the md5 hash of all the registered schemas classes with their versions.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_registered_schemas_version_hash() {
+		$schemas = array_merge( $this->get_registered_table_schemas(),  $this->get_registered_field_schemas() );
+
+		$versions = [];
+		foreach( $schemas as $schema ) {
+			// Skip if not an Interface of Table or Field.
+			if ( ! $schema instanceof Table_Schema_Interface && ! $schema instanceof Field_Schema_Interface ) {
+				continue;
+			}
+
+			$class_name = get_class( $schema );
+			$constant_name = $class_name . '::SCHEMA_VERSION';
+
+			// Skip if the version constant is not defined.
+			if ( ! defined( $constant_name ) ) {
+				continue;
+			}
+
+			$versions[ $class_name ] = constant( $constant_name );
+		}
+
+		return md5( json_encode( $versions ) );
```

Comment:
Not likely to be a problem, but we may want to sort the array to ensure the order doesn't vary and cause false hash changes.

Reply:
Resolved.